File: src/main/java/ac/dnd/server/admission/domain/AdmissionRepository.java
package ac.dnd.server.admission.domain;

import java.util.List;
import java.util.Optional;

import ac.dnd.server.admission.domain.model.ApplicantData;
import ac.dnd.server.admission.domain.model.EventData;
import ac.dnd.server.admission.domain.model.EventsData;
import ac.dnd.server.common.support.EventStatus;

public interface AdmissionRepository {
	Long saveEvent(final EventData event);

	List<EventData> getEvents();

	Optional<ApplicantData> findAdmissionByEventIdAndNameAndEmail(
		Long eventId,
		String name,
		String email
	);

	EventsData findByStatusIn(List<EventStatus> statuses);

}

File: src/main/kotlin/ac/dnd/server/admission/domain/AdmissionRepository.kt
package ac.dnd.server.admission.domain

import ac.dnd.server.admission.domain.enums.EventStatus
import ac.dnd.server.admission.domain.model.ApplicantData
import ac.dnd.server.admission.domain.model.ApplicantStatusData
import ac.dnd.server.admission.domain.model.EventData
import ac.dnd.server.admission.domain.model.EventsData

interface AdmissionRepository {
    fun saveEvent(event: EventData): Long

    fun getEvents(): List<EventData>

    fun findAdmissionByEventIdAndNameAndEmail(
        eventId: Long,
        name: String,
        email: String
    ): ApplicantData?

    fun findApplicantStatusByEventIdAndNameAndEmail(
        eventId: Long,
        nameBlindIndex: String,
        emailBlindIndex: String
    ): ApplicantStatusData?

    fun findByStatusIn(statuses: List<EventStatus>): EventsData
}
